### PR TITLE
0.1 release prep

### PR DESCRIPTION
**Pull request by twilmes**
* Release date set in change logs
* mvn release:prepare run which bumped version 0.1.0, committed, tagged, and then bumped to the new dev version 0.1.1-SNAPSHOT and committed

### CODE REVIEW

**Comment by sjudeng** [comment]
What would be the process if the 0.1.0 release vote fails and changes are required? Would that work happen on top of these commits (which would then have project version of 0.1.1-SNAPSHOT)?

**Comment by twilmes** [comment]
The maven release plugins bump from 0.1.0 to 0.1.1-SNAPSHOT is a bit problematic in the case where further changes are required. We could rollback to the RC2 tag, force a push to 0.1 on that and do any further work after that point.  In the past, I've used the versions plugin to do the bump to the release version, and then later on, the bump to the new SNAPSHOT after the release is squared away.

File: CHANGELOG.asc [hunk 1]
```diff
@@ -3,8 +3,8 @@
 Release Notes
 -------------
 
-Version 0.1.0 (TBD)
-~~~~~~~~~~~~~~~~~~~
+Version 0.1.0 (Release Date: April 11, 2017)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (mbrukman) [inline CHANGELOG.asc]: Can you please file an issue to cleanup the headers here to use the new headings format? Feel free to assign it to me.

File: docs/changelog.txt [hunk 2]
```diff
@@ -3,8 +3,8 @@
 Release Notes
 -------------
 
-Version 0.1.0 (TBD)
-~~~~~~~~~~~~~~~~~~~
+Version 0.1.0 (Release Date: April 11, 2017) 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (mbrukman) [inline docs/changelog.txt]: Same comment here as in `CHANGELOG.asc`.

**Comment by ptgoetz** [comment]
@mbrukman That's kind of non-standard practice. What I usually do when an RC vote fails is (assuming maven release plugin):

1. Delete the tag in the local and remote repos.
2. `git revert` the two commits made by the maven release plugin.

File: CHANGELOG.asc (hunk 1 above)
Comment (twilmes) [inline CHANGELOG.asc]: Good catch, created #211.

**Comment by twilmes** [comment]
Yes, agreed, a revert is the way to go so we don't lose the history.

**Comment by mbrukman** [comment]
OK, so we have the following options in case a vote fails:

* force-push to the branch and rewrite history
* `git revert`
* work in a local branch and push only when vote passes

I'm in favor of the second or third, I prefer not to set a precedent that we rewrite history, it should be moving forward. The second option has the benefit of recording what happened, including the reverts, the third one seems like it's inviting things to be done outside of the repo, so I agree that `git revert` is a good style to follow.

**Comment by twilmes** [comment]
Thanks for the quick feedback guys.  I'll take note of the `git revert` in case of vote failure and capture that in the updated instructions after we complete the release.

**Pull request merged**